module/importer/src/Infrastructure/Provider/SourceTypeProvider.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types = 1);
+
+namespace Ergonode\Importer\Infrastructure\Provider;
+
+/**
+ */
+class SourceTypeProvider
+{
+    /**
+     * @var string[]
+     */
+    private array $types;
+
+    /**
+     * @param string ...$classes
+     *
+     * @throws \ReflectionException
+     */
+    public function __construct(string ...$classes)
+    {
+        $this->types = [];
+        foreach ($classes as $class) {
+            $type = (new \ReflectionClass($class))->getConstant('TYPE');
```

Comment:
It seems like a hack.
Are you expecting here a class of certain interface? If so it could just contain the static method `::getType()`?

Maybe it'd be good jut to inject this TYPE property(however obtained) to the provider directly? Cons are that you have compile time validation(not runtime) and efficiency boost as it only happens once during the building of the container.